apps/gateway/test/champions_test.exs

```diff
@@ -424,6 +428,142 @@ defmodule Gateway.Test.Champions do
     end
   end
 
+  describe "afk rewards" do
+    test "winning battles increments the afk rewards", %{socket_tester: socket_tester} do
+      {:ok, user} = Users.register("AfkRewardsUser")
+
+      # Check that the initial afk reward rates are all 0
+      assert Enum.all?(user.afk_reward_rates, fn rate -> rate.rate == 0 end)
+
+      # Get initial afk rewards
+      SocketTester.get_afk_rewards(socket_tester, user.id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{response_type: {:afk_rewards, %AfkRewards{afk_rewards: initial_afk_rewards}}}
+
+      # Check that the initial afk rewards are all 0
+      assert Enum.all?(initial_afk_rewards, fn reward -> reward.amount == 0 end)
+
+      # Set up a powerful team to win a level with the user. That should increment the afk rewards rates
+      Enum.each(user.units, fn unit ->
+        GameBackend.Units.update_unit(unit, %{level: 9999})
+      end)
+
+      [campaign_progression | _] = user.campaign_progresses
+      level_id = campaign_progression.level_id
+      SocketTester.fight_level(socket_tester, user.id, level_id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{
+        response_type: {:battle_result, _ = battle_result}
+      }
+
+      assert battle_result.result == "win"
+
+      # Get advanced user
+      SocketTester.get_user(socket_tester, user.id)
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{} = advanced_user}}
+
+      # Check that the gold and gems afk rewards rates are now greater than the initial rewards
+      gold_currency_id = Currencies.get_currency_by_name!("Gold").id
+      gems_currency_id = Currencies.get_currency_by_name!("Gems").id
+
+      assert Enum.any?(advanced_user.afk_reward_rates, fn rate ->
+               rate.currency_id == gold_currency_id && rate.rate > 0
```

Comment:
You can also check the currency by name and avoid using a query here, something like:
`rate.currency.name == "Gold" && rate.rate > 0`

Reply:
The reason for this is that an `AFKRewardRate` has no currency field, only the corresponding `currency_id`. That's why I'm using the id to compare.

Reply:
I found sort of a workaround for this, tell me what you think about it

Reply:
aaaahh okey, that's true the struct `AfkRewardRate ` doesn't have this field in the `gateway.proto` file